packages/react-components/react-carousel/library/src/components/useEmblaCarousel.ts

```diff
@@ -142,22 +189,27 @@ export function useEmblaCarousel(
     }
   });
 
+  const handleIndexChange = React.useCallback(() => {
+    const newIndex = emblaApi.current?.selectedScrollSnap() ?? 0;
+    const slides = emblaApi.current?.slideNodes();
+    const actualIndex = emblaApi.current?.internalEngine().slideRegistry[newIndex][0] ?? 0;
+
+    // We set the active or first index of group on-screen as the selected tabster index
+    slides?.forEach((slide, slideIndex) => {
+      setTabsterDefault(slide, slideIndex === actualIndex);
+    });
+    setActiveIndex(newIndex);
+  }, [setActiveIndex]);
+
+  const handleAutoplayIndexChange = useEventCallback(() => {
+    handleIndexChange();
+    const newIndex = emblaApi.current?.selectedScrollSnap() ?? 0;
+    onAutoplayIndexChange?.(null, { event: null, type: 'autoplay', index: newIndex });
+  });
+
   const viewportRef: React.RefObject<HTMLDivElement> = React.useRef(null);
+  const currentElementRef = React.useRef<HTMLDivElement | null>();
```

Comment:
We don't use this ref anywhere, correct? Let's restore `currentElement`, so it won't leak the scope of `useMemo()`.